app/models/internal/alerts/AlertEvaluationResult.java

```diff
@@ -43,13 +44,23 @@
         )
 )
 public interface AlertEvaluationResult {
-
-    // XXX(cbriones): This is likely to change once we actually integrate alerts with the job scheduling code.
+    /**
+     * The series name at the time of evaluation.
+     * <p>
+     * This can be used to distinguish cases where the underlying query changed
+     * between successive evaluations.
+     *
+     * @return the series name.
+     */
+    String getSeriesName();
```

Comment:
IIUC: we're worried about getting a result back and being confused about what it's a result _for_, because the alert changed in the meantime?

If that's the case: have you considered sticking the entire query in here, rather than just the series-name? That would seem maximally informative about what actually got run.
I don't have a strong preference here, though -- I don't expect it to matter much in practice.

Reply:
Yes that's correct. I suppose there could be other ambiguous cases as well (what's the series name for an arithmetic query?). But like you mention I don't expect it matter too much and I was looking for some minimal solution since this value is going to be stored twice a minute in the worst case.

If this did not suffice, I'd rather store some smaller value derived from the query (like a checksum) than the actual query itself.